pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -208,6 +216,37 @@ func watchCerts(ctx context.Context, certsDirs []string, watchFileEventsFn watch
 	watchFileEventsFn(ctx, fw.Event, minDelay, updateFunc)
 }
 
+func certsExist(certs []CertSource, interval time.Duration) {
+	for _, cert := range certs {
+		for {
```

Comment:
This is a potentially dead loop, if somehow cert file is not there, say, CA is down, then it would take forever for the test to be finished. What about increasing the retry time but still let this piece of code to be able to exit.

Reply:
A waitgroup with a timeout may be appropriate (see https://stackoverflow.com/a/32843750)

Reply:
Thanks for the pointer. I will improve it.